compiler/plugins/target/LLVMCPU/ResolveCPUAndCPUFeatures.cpp

```diff
@@ -0,0 +1,163 @@
+    std::string_view loggingUnspecifiedTargetCPU) {
+  if (!cpu.empty() || !cpuFeatures.empty()) {
+    return true;
+  }
+  if (loggingUnspecifiedTargetCPU.empty()) {
+    return false;
+  }
+  if (triple.isX86()) {
+    // On X86, one usually talks in terms of CPU, not individual CPU features.
+    llvm::errs() << R"MSG(
+error: Please pass --iree-llvmcpu-target-cpu (or alternatively, --iree-llvmcpu-target-cpu-features, but this message is tailored to the current target architecture, x86, where --iree-llvmcpu-target-cpu is used more often).
+
+Examples:
+
+    --iree-llvmcpu-target-cpu=host
+        Target the host CPU. The generated code will have optimal performance on the host CPU but will crash on other CPUs not supporting the same CPU features.
+
+    --iree-llvmcpu-target-cpu=generic
+        Target a generic CPU for the target architecture. The generated code will have poor performance unless CPU features are also specified by --iree-llvmcpu-target-cpu-features.
+
+    --iree-llvmcpu-target-cpu=somecpu
+        Target the specified `somecpu` (see accepted values below). The generated code will have optimal performance on the specified CPU, but will crash on older CPUs and will have suboptimal performance on newer CPUs.
+
+Accepted CPU values:
+)MSG";
+    llvm::SmallVector<llvm::StringRef> allAcceptedCpus;
+    llvm::X86::fillValidCPUArchList(allAcceptedCpus, /*Only64Bit=*/true);
+    llvm::interleaveComma(allAcceptedCpus, llvm::errs());
+    llvm::errs() << "\n";
+  } else {
+    // Outside of X86, one usually talks in terms of CPU features, but people
+    // should also be aware of the possibility of specifying a CPU.
+    llvm::errs() << R"MSG(
+error: Please pass --iree-llvmcpu-target-cpu-features and/or -iree-llvmcpu-target-cpu. Outside of x86, --iree-llvmcpu-target-cpu-features is the preferred way of enabling CPU features.
+
+Examples:
+
+    --iree-llvmcpu-target-cpu-features=+somefeature1,+somefeature2,...
+        Enable the comma-separated list of CPU features (each prefixed with a + sign).
+
+    --iree-llvmcpu-target-cpu=host
+        Target the host CPU. The generated code will have optimal performance on the host CPU but will crash on other CPUs not supporting the same CPU features.
+
+    --iree-llvmcpu-target-cpu=generic
+        Target a generic CPU for the target architecture. The generated code will have poor performance unless CPU features are also specified by --iree-llvmcpu-target-cpu-features.
+
+    --iree-llvmcpu-target-cpu=somecpu
+        WARNING: outside of x86, this flag may not actually populate CPU features. It may only serve to select the instruction scheduling model.
+)MSG";
+  }
+
+  if (loggingUnspecifiedTargetCPU == "fatal") {
+    // I know we shouldn't do that in library code. But this is a non-default
+    // flag, and it's going to be very useful to check that we got all instances
+    // fixed before enabling the warning by default.
+    exit(EXIT_FAILURE);
+  }
```

Comment:
Yeah, let's not merge with `exit(EXIT_FAILURE)`, even for a non-default flag. We'll need to propagate the error somehow.